### DISCUSSION TITLE
Admin event keep being sent, even if the create/update user got an error

### DISCUSSION THREAD

**Discussion by satriahu** [keycloak/keycloak#8544] [Q&A] [answered]
Hello.
I just want to ask. Is it expected that the admin event got sent after create/update user in admin console, even if the action got an error?
I've checked the source code in keycloak, it does. while deleting user does not.

It kinda bothers me, because I am developing an auth service and using this create/update event listener to do something else. and it got bugs, because the admin event is sending false events when got error.

Thanks. Have a nice day.

<img width="899" alt="Screen Shot 2021-10-08 at 12 23 45" src="https://user-images.githubusercontent.com/81661682/136503092-e9ec2d6c-a735-4ec8-9e79-973bc5c0bb40.png">

<img width="1345" alt="Screen Shot 2021-10-08 at 12 20 42" src="https://user-images.githubusercontent.com/81661682/136503075-ae6b0cbf-ccb7-4ebe-a220-cd85de2e9a7f.png">

**Comment by stianst** [comment, marked as answer]
We're looking at improving the events quite a bit, but there is a simple interim solution here, which is basically just moving the adminEvent.operation call to after the tx has been committed. Could you open a bug for this, and perhaps even send a PR?

**Reply by satriahu** [reply]
thanks
I'll send a PR and the ticket

https://issues.redhat.com/browse/KEYCLOAK-19602
https://github.com/keycloak/keycloak/pull/8626
